src/main/java/UseCase/RegisterArtistInteractor.java

```diff
@@ -0,0 +1,41 @@
+package UseCase;
+
+import Database.userAccessInterface;
+import Database.userDsData;
+import Entities.User;
+import Entities.userFactory;
+import InputBoundary.RegisterArtistInputBoundary;
+import InputData.RegisterArtistInputData;
+import OutputBoundary.loginOutputBoundary;
+
+public class RegisterArtistInteractor implements RegisterArtistInputBoundary{
+    private final loginOutputBoundary registerPresenter;
+    private final userFactory userFactory;
+
+    userAccessInterface allUsers;
+
+    public RegisterArtistInteractor(loginOutputBoundary registerPresenter, userAccessInterface allUsers,
+                                    userFactory userFactory){
+        this.allUsers = allUsers;
+        this.registerPresenter =registerPresenter;
+        this.userFactory = userFactory;
+    }
+    @Override
+    public void register(RegisterArtistInputData registerInputData){
+        if (allUsers.exists(registerInputData.getUsername())){
+            //TODO: prepare user exists failure view
+        } else if (registerInputData.getPassword() == "") {
+            //TODO: prepare no password failure view
+        } else if (registerInputData.getUsername() == "") {
+            //TODO: prepare no username failure view
+        }
+
+        //create artist user
+        User user;
+        user = this.userFactory.createArtistUser(registerInputData.getArtistName(),registerInputData.getUsername(),
```

Comment:
Lines 34 and 35 can be merged.

Reply:
they r now merged. They were separate before because interactor was shared.